platform/darwin/src/MGLLight.h

```diff
@@ -0,0 +1,66 @@
+#import <CoreGraphics/CoreGraphics.h>
+
+#import "MGLFoundation.h"
+#import "MGLStyleValue.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, MGLLightAnchorType) {
+    MGLLightAnchorTypeMap,
+    MGLLightAnchorTypeViewport
+};
+
+MGL_EXPORT
+@interface MGLLight : NSObject
+
+/**
+ `lightAnchorType` property.
+ */
+@property (nonatomic) MGLStyleValue<NSValue *> *lightAnchorType;
+
+/**
+ Values describing animated transitions to `lightAnchorType` property.
+ */
+@property (nonatomic) MGLTransition lightAnchorTypeTransition;
+
+
+/**
+ Position.
+ */
+@property (nonatomic) MGLPosition position;
+
+/**
+ Values describing animated transitions to `position` property.
+ */
+@property (nonatomic) MGLTransition positionTransiton;
+
+
+/**
+ */
+@property (nonatomic) MGLStyleValue<UIColor *> *color;
+
+/**
+ Values describing animated transitions to `color` property.
+ */
+@property (nonatomic) MGLTransition colorTransiton;
+
+
+/**
+ Specifies the receiver's intensity.
+ */
+@property(nonatomic) MGLStyleValue<NSNumber *> *intensity;
+
+/**
+ Values describing animated transitions to `intensity` property.
+ */
+@property (nonatomic) MGLTransition intensityTransition;
+
+@end
+
+@interface NSValue (MGLLightAdditions)
+
++ (instancetype)valueWithMGLLightAnchorType:(MGLLightAnchorType)lightAnchorType;
+
```

Comment:
In addition to `-valueWithMGLLightAnchorType:`, there should be a read-only `MGLLightAnchorTypeValue` property for converting in the other direction.